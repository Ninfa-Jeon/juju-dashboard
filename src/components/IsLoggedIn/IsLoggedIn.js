import React from "react";
import { useSelector } from "react-redux";
import { isLoggedIn } from "app/selectors";

const IsLoggedIn = ({ children }) => {
  const userIsLoggedIn = useSelector(isLoggedIn);
  const visitURL = useSelector(state => {
    if (!userIsLoggedIn) {
      const root = state.root;
      if (root && root.visitURL) {
        return root.visitURL;
      }
    }
  });

  if (!userIsLoggedIn) {
    return (
      <div>
        Please{" "}
        <a href={visitURL} rel="noopener noreferrer" target="_blank">
          log in
        </a>
      </div>
    );
  }
  return children;
};

export default IsLoggedIn;
