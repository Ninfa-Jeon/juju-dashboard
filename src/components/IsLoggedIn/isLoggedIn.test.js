import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mount } from "enzyme";

import IsLoggedIn from "./IsLoggedIn";

const mockStore = configureStore([]);

describe("IsLoggedIn", () => {
  it("renders a login UI if the user is not logged in", () => {
    const store = mockStore({
      root: {}
    });
    const wrapper = mount(
      <Provider store={store}>
        <IsLoggedIn>I should not be rendered</IsLoggedIn>
      </Provider>
    );
    expect(wrapper.find("IsLoggedIn").text()).toBe("Please log in");
    expect(wrapper).toMatchSnapshot();
  });

  it("renders it's children if the user is logged in", () => {
    const store = mockStore({
      root: {
        controllerConnection: {},
        bakery: {}
      }
    });
    const wrapper = mount(
      <Provider store={store}>
        <IsLoggedIn>Hello world!</IsLoggedIn>
      </Provider>
    );
    expect(wrapper.find("IsLoggedIn").text()).toBe("Hello world!");
    expect(wrapper).toMatchSnapshot();
  });
});
