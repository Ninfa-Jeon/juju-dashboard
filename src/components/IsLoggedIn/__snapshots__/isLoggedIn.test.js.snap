// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsLoggedIn renders a login UI if the user is not logged in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IsLoggedIn>
    <div>
      Please
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        log in
      </a>
    </div>
  </IsLoggedIn>
</Provider>
`;

exports[`IsLoggedIn renders it's children if the user is logged in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IsLoggedIn>
    Hello world!
  </IsLoggedIn>
</Provider>
`;
